async_fuse/src/session.rs

```diff
@@ -157,6 +163,9 @@ impl Session {
                     let fs = self.filesystem.clone();
                     let sender = pool_sender.clone();
                     Task::spawn(async move {
+                        // This task will run in parallel, store the buffer index to avoid data race
+                        let buf_idx = idx;
```

Comment:
Why is "idx" not moved inside the async block?

Reply:
you are right, idx will be copied as soon as the task is spawned.